### PR TITLE
Add UUID fetch method to player

### PR DESCRIPTION
**Pull request by democat3457**

### CODE REVIEW

File: CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/player/MCPlayer.java [hunk 1]
```diff
@@ -39,6 +39,14 @@ public String getName() {
         return player.getName();
     }
     
+    @Override
+    public String getUUID() {
+        String uuid = EntityPlayer.getUUID(player.getGameProfile()).toString().toLowerCase();
+        if (uuid != null && !uuid.equals(""))
```
Comment (kindlich) [inline CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/player/MCPlayer.java]: uuid can never be `null` here because if `EntityPlayer.getUUID(player.getGameProfile())` were to return `null` you'd already get a NPE in line 44.

Maybe use something like

```java
UUID uuid = EntityPlayer.getUUID(player.getGameProfile());
if(uuid == null) {
    uuid = EntityPlayer.getOfflineUUID(player.getName());
}
if(uuid == null) {
    CrafttweakerAPI.logError("Could not get UUID for player " + getName());
    return "";
}

return uuid.toString().toLowerCase();
```

Also, since I don't know these vanilla methods that well, can `player.getGameProfile` return `null`, and if it can, will `getUUID(null)` throw an error?

**Review by kindlich — CHANGES_REQUESTED** [review]
Just some questions added to the MCPlayer implementation ^^
If they are answered, it should be ready to merge

**Comment by friendlyhj** [comment]
Oh, btw, I made a CrT addon recently, which adds the method. But I use `Entity::getUniqueID` to implement. 

If the PR is merged, I should make a change to solve potential conflict.

I wonder if there is a ZenExpansion, which has the homonymous method that ZenClass has, it will use which method?

**Comment by democat3457** [comment]
@friendlyhj I'd think it uses the ZenExpansion's method because it'd override the existing class but I'm not sure exactly how ZenExpansions work. Also I'm pretty sure `Entity#getUniqueID` is different from the expected GameProfile UUID.

**Comment by kindlich** [comment]
@democat3457 @friendlyhj 
The difference between a method and an expansion method is, that an expansion method is a method for a class that you have not written yourself, but expands an existing class.

So friendlyhj adds `getUUID` to IPlayer, even though his mod cannot modify CrT's source code.
On the other hand @democat3457 adds method directly to CrT's source code through this PR.

From what I see, the methods should not overlap, since this PR adds them as getter and friendlyhj uses a method.
So basically
```zs
player.uuid //calls the getter, added by this PR
player.getUUID() //calls the method from friendlyhj
```

**Review by kindlich — APPROVED** [review]
